backend/backup_media.js

```diff
@@ -0,0 +1,171 @@
+    try {
+      dimensions = sizeOf(outputPath);
+    } catch (e) {
+    }
+  }
+  let line = [
+    itemType, itemId, mediaType, mediaSrc, outputFilename, dimensions.width, dimensions.height
+  ];
+  fs.appendFileSync(config.dbFile, line.join(',') + '\n');
+};
+
+const processBatch = async (offset) => {
+  const result = await connection.promise().query(
+    'select * from `content_source` where (`source` like "%<img%" or `source` like "%<video%") and date(created_at) > "' + START_DOWNLOADING_FROM_DATE + '" limit ' + offset + ',' + BATCH_SIZE
+  );
+  if (!result[0] || !result[0].length) {
+    return false;
+  }
+  for (const entry of result[0]) {
+    let itemType = entry.ref_type;
+    let itemId = entry.ref_id;
+    let entrySource = entry.source;
+    const mediaItems = entrySource.match(/<(img|video|source).+src=['"]([^'"]+)['"]/g);
+    if (!mediaItems) {
+      console.log(`No media items detected: `, entrySource);
+      continue;
+    }
+    console.log(`have ${mediaItems.length} media items`);
+    for (const mediaItem of mediaItems) {
+      const check = mediaItem.match(/<(img|video|source).+src=['"]([^'"]+)['"]/);
+      let mediaSrc = check[2];
+      let mediaType = check[1] === MEDIA_TYPE_IMG ? MEDIA_TYPE_IMG : MEDIA_TYPE_VIDEO;
+      let mediaExtensionMatch = mediaSrc.match(/\.[a-z]{2,}(?:\?.+)?$/);
+      let mediaExtension;
+      if (!mediaExtensionMatch) {
+        if (mediaType === MEDIA_TYPE_IMG) {
+          mediaExtension = '.jpg';
+        } else {
+          mediaExtension = '.mp4';
+        }
+      } else {
+        mediaExtension = mediaExtensionMatch[0].replace(/\?.+?$/, '');
+      }
+      let outputFilename = crypto.createHash('sha256').update(mediaSrc).digest('hex') + mediaExtension;
+      let outputPath = config.dir + '/' + outputFilename;
+      if (fs.existsSync(outputPath)) {
+        console.log(outputPath, 'is already there');
+        saveEntry(mediaType, outputPath, outputFilename, mediaSrc, itemType, itemId);
+      } else {
+        console.log(`Processing media item `, mediaSrc);
+        try {
+          const requestController = new AbortController();
+          const requestTimeoutController = new AbortController();
+          await Promise.race([downloadRequestTimeout(MEDIA_DOWNLOAD_REQUEST_TIMEOUT_MS, mediaSrc, requestController), download(mediaSrc, outputPath, requestTimeoutController)]);
+          console.log(`Done processing media item ${mediaSrc}`);
+          saveEntry(mediaType, outputPath, outputFilename, mediaSrc, itemType, itemId);
+          await sleep(SLEEP_BETWEEN_MEDIA_ITEMS_MS);
+        } catch (e) {
+          console.error(e);
+        }
+      }
+    }
+  }
+  return result[0].length;
+};
+
+(async () => {
+  try {
+    const result = await connection.promise().query(
+      'select count(*) as total from `content_source` where (`source` like "%<img%" or `source` like "%video%") and date(created_at) > "' + START_DOWNLOADING_FROM_DATE + '" '
```

Comment:
`content_source.source` doesn't work here, it contains "source" text before parsing, meaning img and video could be present there as raw urls.

Traversing data in the order of  `content_source.content_source_id`  increase is a good idea, since the latest `content_source_id` could be saved as an offset for the next processing.

I'd suggest to rewrite the query as follows (pseudocode):
```sql

select
cs.content_source_id, coalesce(p.html, c.html) as html
( -- inner query:
   select countent_source_id from content_source 
   where  content_source_id > $OFFSET
   order by content_source_id asc
    limit $BATCH_SIZE 
) cs 
  left join posts p on content_source_id 
  left join comments c on content_source_id

where coalesce(p.html, c.html) like '%img/video%'
```